src/cascadia/WpfTerminalControl/TerminalContainer.cs

```diff
@@ -138,8 +137,10 @@ internal void SetTheme(TerminalTheme theme, string fontFamily, short fontSize)
             var dpiScale = VisualTreeHelper.GetDpi(this);
 
             NativeMethods.TerminalSetTheme(this.terminal, theme, fontFamily, fontSize, (int)dpiScale.PixelsPerInchX);
+            var size = this.CalculateRowsAndColumns(this.TerminalControlSize);
 
-            this.TriggerResize(this.RenderSize);
+            this.Rows = (int)size.rows;

```

Comment:
Why not call Resize(Size) here?

Reply:
Good catch, this actually showed me that I was doing some wrong calculations for the margins (hence the original reason I removed it)